core/Cargo.toml

```diff
@@ -46,7 +46,7 @@ ring = { version = "0.16.9", features = ["alloc", "std"], default-features = fal
 async-std = { version = "1.6.2", features = ["attributes"] }
 base64 = "0.13.0"
 criterion = "0.4"
-libp2p = { path = "../", default-features = false, features = ["mplex", "noise"] }
+libp2p = { path = "..", features = ["full"] }
```

Comment:
Say I want to run the `libp2p-core` unit tests locally, wouldn't it be nicer to only compile the `libp2p` features that I need? Is this use-case worth the additional effort?

Reply:
It is a pain though to keep track of the features you need to activate :/

Plus, with all crates requesting different features, I actually think we have to compile things multiple times to satisfy each combination. If the convention is to enable `full` everywhere, we only need to compile every crate once to and the compiled artifact can be used for every crate's tests[^1].

I think the convenience of just saying "full" is worth it. Locally you have caches of everything so you will most likely just pay in linker time and if you modify your code between test executions, you gotta relink the test binary anyway so I am not sure it will matter in practice.

Can we try it out and if it negatively affects anything, figure out how to make it better?

[^1]: This is an unproved hypothesis.

Reply:
Thanks for elaborating. Using `full` sounds good to me.

Reply:
I changed my mind. Esp. working on `libp2p-core` is quite painful now because every change there invalidates `libp2p-core` artifact which causes a re-compile of every other crate.

Reply:
:+1: mind proposing a patch @thomaseizinger?

Reply:
So I think the best way to resolve this is to actually stop the circular dependency. This will allow for proper caching of the build artifacts. It will also solve the issue I am currently facing with `release-please` (#2902).